cmd/integration/commands/export_db_data.go

```diff
@@ -0,0 +1,584 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"math/big"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/common/length"
+	"github.com/erigontech/erigon-lib/kv"
+	"github.com/erigontech/erigon/core/rawdb"
+	"github.com/erigontech/erigon/eth/consensuschain"
+	"github.com/erigontech/erigon/eth/ethconfig"
+	"github.com/erigontech/erigon/polygon/bor/borcfg"
+	"github.com/erigontech/erigon/polygon/bor/snaptype"
+	"github.com/erigontech/erigon/polygon/heimdall"
+	"github.com/erigontech/erigon/rlp"
+	"github.com/erigontech/erigon/turbo/debug"
+	"github.com/erigontech/erigon/turbo/snapshotsync/freezeblocks"
+)
+
+var cmdExportHeaderTd = &cobra.Command{
+	Use:   "export_header_td",
+	Short: "",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := debug.SetupCobra(cmd, "integration")
+		label := kv.ChainDB
+		db, err := openDB(dbCfg(label, path.Join(datadirCli, label.String())), true, logger)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer db.Close()
+		tx, err := db.BeginRo(cmd.Context())
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer tx.Rollback()
+
+		c, err := tx.Cursor(kv.HeaderTD)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer c.Close()
+
+		out, err := os.Create(outputCsvFile)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		csvWriter := csv.NewWriter(out)
+		if err = csvWriter.Write([]string{"source", "block_num", "block_hash", "td"}); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		var k, v []byte
+		for k, v, err = c.First(); err == nil && k != nil; k, v, err = c.Next() {
+			blockNum := binary.BigEndian.Uint64(k[:8])
+			if blockNum < fromNum {
+				continue
+			}
+
+			if blockNum >= toNum {
+				break
+			}
+
+			var blockHash libcommon.Hash
+			copy(blockHash[8:], k)
+
+			td := new(big.Int)
+			if err := rlp.Decode(bytes.NewReader(v), td); err != nil {
```

Comment:
use `rlp.DecodeBytes` - it will do less allocs. (see: https://github.com/erigontech/erigon/pull/12013 )